tests/measure.test.js

```diff
@@ -0,0 +1,31 @@
+import { test } from 'zinnia:test'
+import { assertEquals } from 'zinnia:assert'
+import { measure } from '../lib/measure.js'
+
+test('measure successful retrieval', async () => {
+  const requests = []
+  const mockFetch = async (url, allOpts) => {
+    requests.push(url)
+    return { ok: true, status: 200 }
+  }
+
+  const result = await measure('https://example.com', 'blobId', mockFetch)
+  assertEquals(result, { retrievalSucceeded: true })
+  assertEquals(requests, [
+    'https://example.com/v1/blobs/blobId'
+  ])
+})
+
+test('measure failed retrieval', async () => {
+  const requests = []
+  const mockFetch = async (url, allOpts) => {
+    requests.push(url)
+    return { ok: false, status: 500 }
+  }
+
+  const result = await measure('https://example.com', 'blobId', mockFetch)
+  assertEquals(result, { retrievalSucceeded: false })
+  assertEquals(requests, [
+    'https://example.com/v1/blobs/blobId'
+  ])
+})
```

Comment:
Please add a test to verify the case when the retrieval throws an error. For example:

```js
  const mockFetch = async (url, allOpts) => {
    requests.push(url)
    throw new Error('FETCH ERROR')
  }
```

Reply:
Resolved in https://github.com/CheckerNetwork/walrus-checker/pull/12/commits/035f2cb6ab1e5445d2d47eff1a47acfa327edc4a